test/common/stats/symbol_table_fuzz_test.cc

```diff
@@ -5,26 +5,42 @@
 #include "test/fuzz/fuzz_runner.h"
 #include "test/fuzz/utility.h"
 
+#include "absl/strings/match.h"
+#include "absl/strings/string_view.h"
+
 namespace Envoy {
 namespace Stats {
 namespace Fuzz {
 
+// Adds a stat-name to the symbol table, discarding it if it ends in ".", and
+// splitting it in two if it's too long (64k bytes).
+//
+// The actual requirement for StatName allows up to 64k "."-separated segments
+// of 64k, but it's simpler to just limit the whole string. I don't think
+// there's that much value in using strings larger than 64k.
+void addSymbol(StatNamePool& pool, SymbolTable& symbol_table, absl::string_view str) {
+  while (absl::EndsWith(str, ".")) {
+    str.remove_suffix(1);
+  }
+
+  if (str.size() >= StatNameMaxSize) {
+    size_t halfway = str.size() / 2;
+    addSymbol(pool, symbol_table, str.substr(0, halfway));
```

Comment:
No need to have this kind of smart logic; if you just cap the string at `StatNameMaxSize`, the fuzzer will be able to do what is implied in this logic if it is helpful to the exploration.

Reply:
I'm changing the symbol-table impl to simply not have this limit, which should simplify fuzzing. It works I think but still not quite ready for review. Will ping again when ready.

Reply:
This is ready for review now.